doc/tutorials/04-lattice_boltzmann/04-lattice_boltzmann_part1.ipynb

```diff
@@ -326,178 +324,27 @@
     "boundary is still an important research topic in the LB community and in combination with point particle coupling not much experience exists. This means: do research on that topic, play around with parameters and figure out what happens.\n",
     "\n",
     "Boundaries are initialized by passing a shape object to the <tt>LBBoundary</tt> class. One way to initialize a wall is:\n",
     "\n",
     "```python\n",
-    "from espressomd import lbboundaries, shapes\n",
-    "wall = lbboundaries.LBBoundary(shape=shapes.Wall(normal=[1, 0, 0], dist=1),\n",
+    "import espressomd.lbboundaries\n",
+    "import espressomd.shapes\n",
+    "\n",
+    "wall = espressomd.lbboundaries.LBBoundary(shape=espressomd.shapes.Wall(normal=[1, 0, 0], dist=1),\n",
     "                               velocity=[0, 0, 0.01])\n",
     "system.lbboundaries.add(wall)\n",
     "```\n",
     "\n",
-    "Note that all used variables are inherited from previous examples. This will create a wall with a surface normal of $(1, 0, 0)$ at a distance of 1 from the origin of the coordinate system in direction of the normal vector. The wall exhibits a slip boundary condition with a velocity of $(0, 0, 0.01)$. For the a no-slip condition, leave out the velocity argument or set it so zero. Please refer to the user guide for a complete list of constraints.\n",
-    "\n",
-    "Currently only the so-called *link bounce back* method is implemented, where the effective hydrodynamic boundary is located midway between two nodes. This is the simplest and yet a rather effective approach for boundary implementation. Using the shape objects distance function, the nodes determine once during initialisation whether they are boundary or fluid nodes."
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "## 4 Drag force on objects\n",
-    "\n",
-    "As a first test, we measure the drag force on different objects in a simulation\n",
-    "box. Under low Reynolds number conditions, an object with velocity $\\vec{v}$\n",
-    "experiences a drag force $\\vec{F}_\\text{D}$ proportional to the velocity:\n",
-    "\\begin{align*}\n",
-    "\t\\vec{F}_\\text{D}=-\\gamma\\vec{v},\n",
-    "\\end{align*}\n",
-    "where $\\gamma$ is denoted the friction coefficient. In general $\\gamma$ is a\n",
-    "tensor thus the drag force is generally not parallel to the velocity. For\n",
-    "spherical particles the drag force is given by Stokes' law:\n",
-    "\\begin{align*}\n",
-    "\t\\vec{F}_\\text{D}=-6\\pi\\eta a\\vec{v},\n",
-    "\\end{align*}\n",
-    "\n",
-    "where $a$ is the radius of the sphere.\n",
-    "In this task you will measure the drag force on a spherical object at rest centered in a square channel with LB and **ESPResSo**."
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "import espressomd\n",
-    "espressomd.assert_features(['CUDA', 'LB_BOUNDARIES_GPU'])\n",
```

Comment:
We probably need an `assert_features` here. Tutorial tests rely on that assertion to know if the test can be run or not (otherwise we would have to define the list of assertions in the test itself, and make sure it is always up-to-date). You might pass the tutorial test in the PR CI pipeline, but as soon as it is merged, the main branch CI will run additional tutorial jobs, including one with an empty myconfig file. Can this tutorial run without any feature?

Reply:
they only depend on ```LB_BOUNDARIES_GPU``` which depends on the external feature ```CUDA```. Of course this tutorial can only be run with a gpu.